flask_backend/app.py

```diff
@@ -2,81 +2,151 @@
 import json
 
 
 app = Flask(__name__)
 CORS(app)
 
-users = { 
-   'users_list' : []
+#app
+
+jobs = { # users before
+   'job_list' : # users_list before
+   [
+      { 
+         'position' : 'janitor',
+         'company' : 'google',
+         'status': 'Waiting to Hear Back',
+         'additional_info': 'i am so excited',
+         '_id': '12345678910ac43185927543',
+      },
+      {
+         'position' : '',
+         'company' : '',
+         'status': '',
+         'additional_info': '',
+         '_id': '000000000000000000000000',
+      }
+ ]
 }
 
-# def random_id_generator():
-#     return uuid.uuid4().hex[:6]
-    
-@app.route('/users', methods=['GET', 'POST','DELETE'])
-def get_users():
-   if request.method == 'GET':
-      search_username = request.args.get('name')
-      search_job = request.args.get('job')
-      if search_username and search_job :
-         #return find_users_by_name_job(search_username, search_job) 
-         users = User().find_by_name_and_job(search_username, search_job)
-      elif search_username  :
-         users = User().find_by_name(search_username)
-      elif search_job  :
-         return find_users_by_job(search_job) 
-      else:
-         users = User().find_all()
-      return {"users_list": users}
-   elif request.method == 'POST':
-      userToAdd = request.get_json()
-      # userToAdd['id'] = gen_random_id() # check for duplicate before appending.. todo
-      # users['users_list'].append(userToAdd)
-
-      # make DB request to add user
-      newUser = User(userToAdd)
-      newUser.save()
-      resp = jsonify(newUser), 201
-      return resp
+def getJobsFromQuery():
+   search_status = request.args.get('status')
+   search_name = request.args.get('company')
+   if search_status:
+      jobs = Job().find_by_status(search_status)# just need this
+   elif search_name:
+      jobs = Job().find_by_company(search_name)
+   return {"job_list": jobs}                    # make sure this is returning correctly
+
+
+# def editJob(): # can either do a delete and then an add or write a new mongo function that edits the job
+
+#    # we need the old name of the job in order to do a delete, maybe new delete that takes in a name??
+
+
+#    # we then want the new date in order to do a job add
+
+
+#    return
+ 
+ 
+def addJob(): 
+   jobToAdd = request.get_json()
+   jobs['job_list'].append(jobToAdd) 
+   # userToAdd['id'] = gen_random_id()     # check for duplicate before appending.. todo
+   # make DB request to add user
+   newJob = Job(jobToAdd)
+   newJob.save()
+   resp = jsonify(newJob), 201
+   return resp
+
+
+def deleteJob():
+   jobToDelete = request.get_json()
+   resp = jobToDelete.remove()            # DOUBLE CHECK THAT THIS WORKS                  
+   #print(f'resp = {resp}')
+   if resp['n'] == 1:                     # WHAT DOES THIS DO?????
+      return {}, 204
+   #if user.reload() :
+   #   return user
+   else :
+      return jsonify({"error": "job not found"}), 404
```

Comment:
Maybe figure out why line 71 wasnt working

Reply:
I think it is a completely unnecessary line that was used for testing in the previous file that I got it from.